File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/crc32/Crc32Suite.kt
package cryptobench.ciphers.crc32

import cryptobench.core.CipherSuite
import cryptobench.core.HashFunction
import cryptobench.core.SuiteResult
import cryptobench.verify.distinctDigests
import cryptobench.verify.stableDigest

class Crc32Suite : CipherSuite {
    override fun name(): String = "crc32"

    override fun category(): String = "hash"

    override fun run(): SuiteResult {
        val hash: HashFunction = Crc32Hash()
        val samples = Crc32Vectors.samples()
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in samples) {
            if (stableDigest(hash, sample)) passed++ else failed++
        }
        for (other in samples) {
            if (distinctDigests(hash, samples[0], other)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}

File: fixtures/scale/dart/large/lib/ciphers/crc32/crc32_suite.dart
import '../../core/cipher_suite.dart';
import '../../core/hash_function.dart';
import '../../core/suite_result.dart';
import '../../verify/determinism.dart';
import 'crc32_hash.dart';
import 'crc32_vectors.dart';

class Crc32Suite implements CipherSuite {
  @override
  String name() => 'crc32';

  @override
  String category() => 'hash';

  @override
  SuiteResult run() {
    final HashFunction hash = Crc32Hash();
    final samples = crc32Samples();
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in samples) {
      if (stableDigest(hash, sample)) passed++; else failed++;
    }
    for (final other in samples) {
      if (distinctDigests(hash, samples[0], other)) passed++; else failed++;
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}
